packages/opentelemetry-api/src/diag/consoleLogger.ts

```diff
@@ -0,0 +1,111 @@
+          // Some environments only expose the console when the F12 developer console is open
+          let theFunc = console[funcName];
+          if (!theFunc) {
+            // Not all environments support all functions
+            theFunc = console.log;
+          }
+
+          // One last final check
+          if (theFunc && typeof theFunc === 'function') {
+            return theFunc.apply(console, orgArguments);
+          }
+        }
+      };
+    }
+
+    for (let lp = 0; lp < consoleMap.length; lp++) {
+      const name = consoleMap[lp].n;
+      let consoleFunc = consoleMap[lp].c;
+      if (console && !console[consoleFunc]) {
+        consoleFunc = 'log';
+      }
+      this[name] = _consoleFunc(consoleFunc);
+    }
+  }
+
+  /** Log a terminal situation that would cause the API to completely fail to initialize,
+   * if this type of message is logged functionality of the API is not expected to be functional.
+   */
+  public terminal!: DiagLogFunction;
+
+  /** Log a critical error that NEEDS to be addressed, functionality of the component that emits
+   * this log detail may non-functional. While the overall API may be.
+   */
+  public critical!: DiagLogFunction;
+
+  /** Log an error scenario that was not expected and caused the requested operation to fail. */
+  public error!: DiagLogFunction;
+
+  /** Log a warning scenario to inform the developer of an issues that should be investigated.
+   * The requested operation may or may not have succeeded or completed.
+   */
+  public warn!: DiagLogFunction;
+
+  /** Log a general informational message, this should not affect functionality.
+   * This is also the default logging level so this should NOT be used for logging
+   * debugging level information.
+   */
+  public info!: DiagLogFunction;
+
+  /** Log a general debug message that can be useful for identifying a failure.
+   * Information logged at this level may include diagnostic details that would
+   * help identify a failure scenario. Useful scenarios would be to log the execution
+   * order of async operations
+   */
+  public debug!: DiagLogFunction;
+
+  /** Log a detailed (verbose) trace level logging that can be used to identify failures
+   * where debug level logging would be insufficient, this level of tracing can include
+   * input and output parameters and as such may include PII information passing through
+   * the API. As such it is recommended that this level of tracing should not be enabled
+   * in a production environment.
+   */
+  public trace!: DiagLogFunction;
+
+  /** Log a general informational message that should always be logged regardless of the
+   * current {@Link DiagLogLevel) and configured filtering level. This type of logging is
+   * useful for logging component startup and version information without causing additional
+   * general informational messages when the logging level is set to DiagLogLevel.WARN or lower.
+   */
+  public forcedInfo!: DiagLogFunction;
```

Comment:
Not really, the end-user should not be using the DiagLogger as it's for the API, SDK or Contrib packages as it's a diagnostic logger for OT to use... And the DiagLogger will/should only be used by end-users when during development to diagnose why something isn't working.

Reply:
I really dont like this, i don't think any of the component should be able to force log something. If i have a specific level in place to make diagostics i don't expect having log coming where i explicitly dont want them, even if its to give me some informations (that in my case i dont want).

Reply:
By default the logger is a Noop, and the diagnostic logger is for us to help end-users (and most likely component implementers) diagnose and find why something in OT is not working. Based on history it is extremely useful to not just "know" what version of the component they are using but also to have it validated within any log / output from this that they will provide (because we will be requesting this). Which is also why I added the description of when this should be used.

Reply:
I had the same concern and it was addressed at the SIG. By default nothing will be logged, even the "forcedInfo." Maybe it is a bad name and we should call it "startupInfo" or something one level higher than info? Then users can decide if they want all info, or startup info only.

Reply:
> Then users can decide if they want all info, or startup info only.

I don't see how they can do this since there are no level for this "forcedInfo" (or "startupInfo") ? Maybe i'm missing something ?

Reply:
Not exactly the purpose, but I think a reasonable compromise -- so it would be between WARN and INFO.. 
Potentially called "STARTUPINFO"

Reply:
> > Then users can decide if they want all info, or startup info only.
> 
> I don't see how they can do this since there are no level for this "forcedInfo" (or "startupInfo") ? Maybe i'm missing something ?

The current implementation logs it as INFO level even if the configured log level is WARN or ERROR. I'm suggesting we add a STARTUPINFO level which would live between INFO and WARN instead.

Reply:
I fine with this

Reply:
Personally, I don't really see the value in INFO level logs in the diagnostic logger. During normal operation the sdk/api shouldn't log